src/eradiate/test_tools.py

```diff
@@ -71,3 +75,142 @@ def missing_artefact(filename: PathLike) -> None:
 
     else:
         raise ValueError(f"unsupported file extension {filename.suffix}")
+
+
+def assert_rmse(
+    result: list, reference: list, rmse_threshold: float = 1e-4
+) -> (bool, float):
+    """
+    Assert if a test passes or fails based on the RMSE between the test result
+    and the reference.
+
+    Parameters
+    ----------
+
+    result : array
+        Result of the test execution
+
+    reference : array
+        Reference result to be compared to
+
+    rmse_threshold : float
+        RMSE threshold value for the assertion. Should be provided, since
+        the RMSE is a scale dependent measure.
+
+    Raises
+    ------
+    ValueError
+        Result and reference do not have the same shape
+
+    Notes
+    -----
+    `result` and `reference` must have the same shape.
+    """
+    if np.shape(result) != np.shape(reference):
+        raise ValueError(
+            f"Result and reference do not have the same shape!\n"
+            f"Got: {np.shape(result)}, {np.shape(reference)}"
+        )
+
+    result_flat = np.array(result).flatten()
+    ref_flat = np.array(reference).flatten()
+
+    rmse = np.linalg.norm(result_flat - ref_flat) / np.sqrt(len(ref_flat))
+
+    return rmse <= rmse_threshold, rmse
+
+
+def assert_chi2(reference, result, significance=0.05):
+    from mitsuba.core.math import chi2
+    from mitsuba.python.math import rlgamma
+
+    histo_bins = np.linspace(reference.min(), reference.max(), 20)
+    histo_ref = np.histogram(reference, histo_bins)[0]
+    histo_res = np.histogram(result, histo_bins)[0]
+
+    # sorting both histograms following the ascending frequencies in
+    # the reference. Algorithm from:
+    # https://stackoverflow.com/questions/9764298/how-to-sort-two-lists-which-reference-each-other-in-the-exact-same-way
+    histo_ref_sorted, histo_res_sorted = zip(
+        *sorted(zip(histo_ref, histo_res), key=lambda x: x[0])
+    )
+
+    chi2val, dof, pooled_in, pooled_out = chi2(histo_res_sorted, histo_ref_sorted, 5)
+
+    p_value = 1 - rlgamma(dof / 2.0, chi2val / 2.0)
+
+    if p_value < significance:
+        passed = False
+    else:
+        passed = True
+
+    return passed, p_value
+
+
+def create_archive_files(path: PathLike, filename: str, data: xr.Dataset):
+    """
+    Create NetCDF output files for regression tests to be archived.
+
+    Parameters
+    ----------
+
+    path : path-like
+        Path of the directory to store the NetCDF file.
+        If the directory does not exist, attempt to create it
+
+    filename: str
+        Filename for result storage
+
+    data: xarray.Dataset
+        Result data to be stored in a NetCDF file
+    """
+
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+    data.to_netcdf(os.path.join(path, filename))
+
+
+def create_regression_test_plots(data: xr.Dataset, filename: PathLike, metric: tuple):
```

Comment:
As previously: maybe something more to-the-point, like (I'll also take the opportunity to fix type hints)
```suggestion
def regression_test_plots(data: xr.Dataset, filename: PathLike, metric: t.Tuple[str, float]):
```